idaes/apps/grid_integration/multiperiod/tests/test_multiperiod.py

```diff
@@ -424,52 +426,153 @@ def test_multi_day_year_stochastic_model(build_multi_day_year_stochastic_model):
+    caplog.clear()
+    with caplog.at_level(idaeslog.WARNING):
+        m = MultiPeriodModel(
+            n_time_points=2,
+            process_model_func=build_flowsheet,
+            linking_variable_func=get_linking_variable_pairs,
+            periodic_variable_func=get_linking_variable_pairs,
+            use_stochastic_build=True,
+            initialization_func=fix_dof_and_initialize,
+            unfix_dof_func=unfix_dof,
+        )
+
+    assert (
+        "A method is provided for get_periodic_variable_pairs. "
+        "build_stochastic_multi_period method does not support periodic "
+        "constraints, so the user needs to add them manually." in caplog.text
     )
 
 
+@pytest.mark.unit
+def test_data_kwargs(caplog, n_time_points=3):
+    # Cover message associated with using model_data_kwargs
+    with caplog.at_level(idaeslog.WARNING):
+        m = MultiPeriodModel(
+            n_time_points=n_time_points,
+            process_model_func=build_flowsheet,
+            linking_variable_func=get_linking_variable_pairs,
+            initialization_func=fix_dof_and_initialize,
+            unfix_dof_func=unfix_dof,
+        )
+
+        data = {
+            0: {
+                "a": 10,
+                "b": 20,
+            },
+            1: {
+                "a": 12,
+                "b": 18,
+            },
+            2: {
+                "a": 15,
+                "b": 22,
+            },
+        }
+
+        m.build_multi_period_model(model_data_kwargs=data)
+
+        assert (
+            f"model_data_kwargs argument is provided, so the flowsheet "
+            f"options are different for different time instances. In this case, "
+            f"the multiperiod model is returned without initialization." in caplog.text
+        )
+
+
+@pytest.mark.unit
+def test_data_kwargs_fail(caplog, n_time_points=2):
+    # Cover warning associated with len(model_data_kwargs) != n_time_points
+    with caplog.at_level(idaeslog.WARNING):
+        m = MultiPeriodModel(
+            n_time_points=n_time_points,
+            process_model_func=build_flowsheet,
+            linking_variable_func=get_linking_variable_pairs,
+            initialization_func=fix_dof_and_initialize,
+            unfix_dof_func=unfix_dof,
+        )
+
+        data = {
+            0: {
+                "a": 10,
+                "b": 20,
+            },
+            1: {
+                "a": 12,
+                "b": 18,
+            },
+            2: {
+                "a": 15,
+                "b": 22,
+            },
+        }
+
+        m.build_multi_period_model(model_data_kwargs=data)
+
+        assert (
+            f"len(model_data_kwargs) != n_time_points.\n "
+            f"len(model_data_kwargs) = {len(data)}\n"
+            f"len(n_time_points) = {n_time_points}\n"
+            f"Check input data for model_data_kwargs argument." in caplog.text
+        )
+
+
 @pytest.mark.unit
 def test_initialization_fail():
     with pytest.raises(
```

Comment:
For consistency with the result of the tests in this file, add a one-line comment explaining the main goal of the test